src/cursor/cur_hs.c

```diff
@@ -380,6 +396,12 @@ __curhs_prev_visible(WT_SESSION_IMPL *session, WT_CURSOR_HS *hs_cursor)
         if (F_ISSET(std_cursor, WT_CURSTD_HS_READ_COMMITTED))
             break;
 
+        /*
+         * If we are using history store cursor and do not have a snapshot, we should set the flag
```

Comment:
This comment is still not great, perhaps something like:

"If we are using a history store cursor and haven't set the WT_CURSTD_HS_READ_COMMITTED flag then we must have a snapshot, assert that we do."

Reply:
Again the same below.

Reply:
Fixed.